src/components/NameValueInput/NameValueInput.tsx

```diff
@@ -34,6 +35,11 @@ function getErrors(errors: FieldErrorsImpl<CheckFormValues | Probe>, name: NameV
   return undefined;
 }
 
+const getStyles = (theme: GrafanaTheme2) => ({
+  addButton: css({ 'margin-top': theme.spacing(1) }),
+  field: css({ 'margin-bottom': 0, 'margin-top': theme.spacing(1) }),
```

Comment:
Nice catch Chris! I removed the `VerticalGroup` and `HorizontalGroup` components in favor of using `stack` and `stackCol` classes, and removed that margin top that made the delete button alignment wrong.

![image](https://github.com/grafana/synthetic-monitoring-app/assets/6271380/81b47e35-d509-4b94-b497-96596f245136)

I do think it would be a good idea to update `grafana/ui` though, so I created this issue for it: https://github.com/grafana/synthetic-monitoring-app/issues/799